app/services/container_dashboard_service.rb

```diff
@@ -133,6 +134,7 @@ def heatmaps
       node_memory_usage = []
 
       metrics.each do |m|
+        next if m.resource.nil? # Metrics are purged asynchronously and might be missing their node
```

Comment:
```m.resource_id``` isnt nil in this situation- its whatever id the node that got deleted had.
meaning I cant filter by m.resource_id.nil?

this metric gets retrieved by the query even though it doesnt belong to a node anymore.  
Theres a time gap in which this metric exists and its resource doesnt. That causes an exception on ```m.resource.name```